src/inmanta/warnings.py

```diff
@@ -0,0 +1,102 @@
+"""
+    Copyright 2020 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+import warnings
+from typing import Dict, List, Mapping, Optional, Type, Union
+
+
+class WarningRule:
```

Comment:
This could use a bit of documentation. I have no idea what it does or why.

Reply:
I'll add some documentation if you decide it should stay in (see my comment above). For now, it's a representation of Python's builtin warnings filter tuple: https://docs.python.org/3.8/library/warnings.html#the-warnings-filter. It indicates what action (ignore, warn, error, ...) should be taken for a warning that matches the rule's message, warning subclass, module and lineno where it occurred.

Reply:
I missed the comment. 

If the filters are used, they can stay, but I don't see where they are applied.

If they are dead code, they should be removed.

Reply:
They are used, but only for a single option, where the rule is just instantiated as `WarningRule(desired_behaviour)` where `desired_behaviour` is one of `default`, `error`, `ignore`. This rule is a catch-all that toggles the desired behaviour for all warnings.
So I wouldn't really call it dead code, but I would call it overengineered for the currently limited application.

Reply:
Perhaps best to keep it simple, but no hard preference.

Reply:
I kept the core of it to keep some abstraction but I simplified and added some documentation.